be/src/io/cache_input_stream.cpp

```diff
@@ -215,28 +223,39 @@ Status CacheInputStream::_populate_to_cache(const int64_t offset, const int64_t
     for (int64_t write_offset_cursor = offset; write_offset_cursor < write_end_offset;) {
         DCHECK(write_offset_cursor % _block_size == 0);
         WriteCacheOptions options{};
+        options.async = _enable_async_populate_mode;
         const int64_t write_size = std::min(_block_size, write_end_offset - write_offset_cursor);
-        Status r = _cache->write_buffer(_cache_key, write_offset_cursor, write_size, src_cursor, &options);
-
-        src_cursor += write_size;
-        write_offset_cursor += write_size;
 
+        SharedBufferPtr sb = nullptr;
+        auto ret = _sb_stream->find_shared_buffer(write_offset_cursor, write_size);
+        if (ret.ok()) {
+            sb = ret.value();
+            auto cb = [sb](int code, const std::string& msg) {
```

Comment:
> Didn't use `sb` in lambda capture?

Yes, because we only need capture the `sb` and hold its reference to avoid it be free before the population finished.

Reply:
what's the relationship between sb/_sb_stream/_cache->write_buffer, it is so implicit here why need to hold the sb which means, the interface can be so easily to be used in the incorrect way.